protocol_rfcs/type-widening.md

```diff
@@ -0,0 +1,143 @@
+# Type Widening
+**Associated Github issue for discussions: https://github.com/delta-io/delta/issues/2623**
+
+This protocol change introduces the Type Widening feature, which enables changing the type of a column or field in an existing Delta table to a wider type.
+
+--------
+
+# Type Widening
+> ***New Section after the [Clustered Table](#clustered-table) section***
+
+The Type Widening feature enables changing the type of a column or field in an existing Delta table
+to a wider type.
+
+The **allowed type changes** are:
+- Integer widening: `Byte` -> `Short` -> `Int` -> `Long`
+- Floating-point widening: `Float` -> `Double`
+- Decimal widening: `Decimal(p, s)` -> `Decimal(p', s')` where `p' >= p` and `p' - s' >= p - s`. `p` and `p'` denote the decimal precision and `s` and `s'` denote the decimal scale.
```

Comment:
I don't think this is quite correct? It allows `s'` less than `s` as long as `p'` is sufficiently larger than `p`. Would it be more rigorous to say: `Decimal(p, s)` -> `Decimal(p + k1, s + k2)` where `k1 >= k2 >= 0` ?